tests/modules/aioseop/test-All_in_One_SEO_Pack-aiospMrtGetUrl.php

```diff
@@ -138,15 +147,191 @@ public static function tearDownAfterClass() {
+			global $post, $post_type, $post_type_object;
+			if ( ! is_object( $post ) || $post->ID !== $this->post_ids[0] ) {
+				$post = get_post( $this->post_ids[0] );
+			}
+			// Set global $post_type with $post->post_type.
+			$post_type = $post->post_type;
+			// Set global $post_type_object = get_post_type_object( $post_type );
+			$post_type_object = get_post_type_object( $post_type );
+			// Switch-case sets global $post = get_post($post_id, OBJECT, 'edit');
+			$post = get_post( $this->post_ids[0], OBJECT, 'edit' );
+
+			// - Operations in All_in_One_SEO_Pack::get_page_snippet_info().
+			// The global $wp_query is still un-initiated/null values.
+			// global $post is set to page being edited. $post = get_post( ID )
+			global $wp_query;
+			// Set $wp_query->is_single = true;
+			$wp_query->is_single = true;
+			// Set $this->is_front_page = false;
+			$this->is_front_page = false;
+			// Set $wp_query->queried_object = $post;
+			// When ! empty( $aioseop_options['aiosp_no_paged_canonical_links'], is when $show_page = false.
+
+			// - Operation in test function.
+			// Would need to be empty
+			//$aioseop_options['aiosp_customize_canonical_links']
+			//$opts['aiosp_custom_link']
+
+			global $aioseop_options;
+			$aioseop_options['aiosp_no_paged_canonical_links']  = false;
+			$aioseop_options['aiosp_customize_canonical_links'] = false;
+			$t01 = $aioseop_class->aiosp_mrt_get_url( $wp_query, true );
+			$t02 = $aioseop_class->aiosp_mrt_get_url( $wp_query, false );
+			$this->assertFalse( $t01, 'All_in_One_SEO_Pack_Sitemap::aiosp_mrt_get_url() has failed on Admin Post-Edit screen. \tests\test-All_in_One_SEO_Pack_test_aiosp_mrt_get_url.php' );
+			$this->assertFalse( $t02, 'All_in_One_SEO_Pack_Sitemap::aiosp_mrt_get_url() has failed on Admin Post-Edit screen. \tests\test-All_in_One_SEO_Pack_test_aiosp_mrt_get_url.php' );
+
+			global $aioseop_options;
+			$aioseop_options['aiosp_no_paged_canonical_links']  = true;
+			$aioseop_options['aiosp_customize_canonical_links'] = true;
+			$t03 = $aioseop_class->aiosp_mrt_get_url( $wp_query, true );
+			$t04 = $aioseop_class->aiosp_mrt_get_url( $wp_query, false );
+			$this->assertFalse( $t03, 'All_in_One_SEO_Pack_Sitemap::aiosp_mrt_get_url() has failed on Admin Post-Edit screen. \tests\test-All_in_One_SEO_Pack_test_aiosp_mrt_get_url.php' );
+			$this->assertFalse( $t04, 'All_in_One_SEO_Pack_Sitemap::aiosp_mrt_get_url() has failed on Admin Post-Edit screen. \tests\test-All_in_One_SEO_Pack_test_aiosp_mrt_get_url.php' );
+		}
+
+		/**
+		 * Override function for go_to edit-posts
+		 *
+		 * (Original) Sets the global state to as if a given URL has been requested.
+		 *
+		 * This sets:
+		 * - The super globals.
+		 * - The globals.
+		 * - The query variables.
+		 * - The main query.
+		 *
+		 * @since 2.4
+		 *
+		 * @see WP_UnitTestCase::go_to()
+		 *
+		 * @param string $url The URL for the request.
+		 */
+		public function go_to_edit_post1( $url ) {
```

Comment:
@EkoJr this looks like a good contender to go into the base class.

Reply:
This is a workaround for a possible defect with `WP_Testcase::go_to()`, and unsure if it is a (un)known bug or intended function. 

Are you sure it's not too early? Plus, there's two different concepts.